platform/android/java/lib/src/org/godotengine/godot/GodotActivity.kt

```diff
@@ -41,6 +41,26 @@ import androidx.fragment.app.FragmentActivity
 import org.godotengine.godot.utils.PermissionsUtil
 import org.godotengine.godot.utils.ProcessPhoenix
 
+import android.os.Build
+import android.os.PowerManager
+import androidx.annotation.RequiresApi
+
+
+
+class GodotThermalWrapper {
+	companion object {
+		public var power_manager: PowerManager? = null
+		@RequiresApi(Build.VERSION_CODES.R)
+		public fun getThermalHeadRoom(forecast: Int): Float {
+			if (Build.VERSION.SDK_INT >= 30) {
+				return power_manager!!.getThermalHeadroom(forecast);
+			} else {
+				return -1.0f;
+			}
+		}
+	}
+}
+
```

Comment:
I moved Thermal changes out of this PR.

I don't think they're ready for inclusion. It's too Android-centric (ignores iOS) and invades main code instead of living in its own class; and it's not clearly defined what its purpose is.

Also, it's stalling Swappy & Pre-Transformed Swapchains which are much higher priority.

This should be ready for review again.

Although... I cannot test it on Android because of #97341